core/trino-main/src/main/java/io/trino/execution/SetViewAuthorizationTask.java

```diff
@@ -66,13 +71,15 @@ public String getName()
         Session session = stateMachine.getSession();
         QualifiedObjectName viewName = createQualifiedObjectName(session, statement, statement.getSource());
         getRequiredCatalogHandle(metadata, session, statement, viewName.getCatalogName());
-        if (!metadata.isView(session, viewName)) {
-            throw semanticException(TABLE_NOT_FOUND, statement, "View '%s' does not exist", viewName);
-        }
+        ViewDefinition view = metadata.getView(session, viewName)
+                .orElseThrow(() -> semanticException(TABLE_NOT_FOUND, statement, "View '%s' does not exist", viewName));
 
         TrinoPrincipal principal = createPrincipal(statement.getPrincipal());
         checkRoleExists(session, statement, metadata, principal, Optional.of(viewName.getCatalogName()).filter(catalog -> metadata.isCatalogManagedSecurity(session, catalog)));
 
+        if (!view.isRunAsInvoker() && !isAllowSetViewAuthorization) {
+            denySetViewAuthorization(viewName.getCatalogName() + '.' + viewName.getSchemaName() + '.' + viewName.getObjectName(), principal);
```

Comment:
Should we throw something else than access denied? Access denied could be misleading and one would try to understand what happened to the the access. Maybe let's say that the feature is disabled.

Reply:
Good idea